fs-storage/src/file_storage.rs

```diff
@@ -62,75 +72,43 @@ where
         + Monoid<V>,
 {
     /// Create a new file storage with a diagnostic label and file path
-    pub fn new(label: String, path: &Path) -> Self {
-        Self {
+    pub fn new(label: String, path: &Path) -> Result<Self> {
+        let time = SystemTime::now();
+        let mut storage = Self {
             label,
             path: PathBuf::from(path),
-            modified: SystemTime::now(),
+            modified: time,
+            written_to_disk: time,
```

Comment:
When do these 2 fields have different values?

Reply:
When a value is set only `modified` is updated to current time. So it becomes `modified > written_to_disk` which shows that mapping is ahead of file.

When `write_fs` or `read_fs` is called it becomes `modified = written_to_disk` to reflect that mapping and file are in sync.

Reply:
I think it's worth to document these invariants as comments to `modified` and `written_to_disk` variables.